src/Symfony/Component/Form/Extension/Core/Type/DateType.php

```diff
@@ -55,6 +55,10 @@ public function buildForm(FormBuilderInterface $builder, array $options)
             throw new InvalidOptionsException('The "format" option must be one of the IntlDateFormatter constants (FULL, LONG, MEDIUM, SHORT) or a string representing a custom format.');
         }
 
+        if (!empty($pattern) && !preg_match('/^([yMd]+)[^yMd]*([yMd]+)[^yMd]*([yMd]+)$/', $pattern)) {
+            trigger_error(sprintf('The DateTime format "%s" is not supported in ICU. Please be aware that the data may not reflect the format.', $pattern), E_USER_WARNING);
```

Comment:
I would throw an `InvalidOptionsException` instead.

Reply:
Ill do that.